paper-api/src/main/java/io/papermc/paper/event/packet/UncheckedSignChangeEvent.java

```diff
@@ -0,0 +1,84 @@
+package io.papermc.paper.event.packet;
+
+import io.papermc.paper.math.BlockPosition;
+import net.kyori.adventure.text.Component;
+import org.bukkit.block.sign.Side;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Unmodifiable;
+import org.jspecify.annotations.NullMarked;
+import java.util.List;
+
+/**
+ * Called when a sign edit packet has been received, but the location at which the sign should be edited
+ * has not yet been checked for the existence of a real sign.
+ * <p>
+ * Cancelling this event will prevent further processing of the sign change, but needs further handling
+ * by the plugin because the local world might be in an inconsistent state.
+ */
+@NullMarked
+public class UncheckedSignChangeEvent extends PlayerEvent implements Cancellable {
```

Comment:
Idk if I like the prefix "unchecked". I get what this is meant to do, but couldn't we just call it `UnknownSignChangeEvent` similar to `PlayerUseUnknownEntityEvent` and only call this in the path of no sign actually being there?

Reply:
I don't like to change it only when no sign is actually there, because when using [UnsafeValues#nextEntityId()](https://jd.papermc.io/paper/1.21.4/org/bukkit/UnsafeValues.html#nextEntityId()), you are guaranteed that your fake entity id is unique and therefore interacting with your fake entity will always call that event.

Contrary to that there might be a sign already placed somewhere, and searching for a free spot to "place" the sign seems kinda whack for something that is not persistent.

As for the name of the event, I am open for suggestions, because that is best name that I could come up taking my reasoning into account